source/server/converters.h

```diff
@@ -26,6 +26,44 @@ inline void trim_trailing_nulls(std::string& s)
       s.end());
 }
 
+enum MatchState {
+  MISMATCH,
+  MATCH,
+  MATCH_OR_ZERO
+};
+
+struct LinkedArraySize {
+  int size = 0;
+  MatchState match_state = MatchState::MISMATCH;
+};
+
+template <size_t N>
+inline LinkedArraySize calculate_linked_array_size(std::array<int, N> sizes, bool allow_optional)
+{
+  auto result = LinkedArraySize{};
+
+  if (std::adjacent_find(sizes.begin() + 1, sizes.end(), std::not_equal_to<>()) == sizes.end()) {
```

Comment:
Why does this start at `sizes.begin() + 1`? Won't this miss the case where the first two elements are non-zero but different? I would think it should start at `sizes.begin()`...

Reply:
Oops. Yeah, there are alternate implementations where you compare everything to the first element, which start with `begin() + 1`. But `adjacent_find` needs to start at the beginning.